pinot-segment-local/src/main/java/org/apache/pinot/segment/local/segment/index/readers/text/LuceneTextIndexReader.java

```diff
@@ -150,10 +155,14 @@ public MutableRoaringBitmap getDocIds(String searchQuery) {
       // be instantiated per query. Analyzer on the other hand is stateless
       // and can be created upfront.
       QueryParser parser = new QueryParser(_column, _analyzer);
+      parser.setAllowLeadingWildcard(true);
```

Comment:
same as above

Reply:
done. better to protect behind the flag for now too.